### DISCUSSION TITLE
CSRF issue in `/admin` after update

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#10140] [Q&A / Support]
In the processing of upgrading a 6.9.6 installation to 10.1.1, one thing has broken and I don't know why, the `/admin` section is showing a 502 error despite having cleared all the caches and emptied all tmp directories. Have anyone seen this error previously?

```bash
curl -I https://localhost/admin --insecure
HTTP/2 403
server: nginx
date: Wed, 25 Aug 2021 06:28:08 GMT
content-type: text/html; charset=UTF-8
vary: Accept-Encoding
set-cookie: PHPSESSID=4f147f4176303c23ebb24a270ebe7e66; path=/; HttpOnly; SameSite=strict
x-debug-exception: Detected%20CSRF%20Attack%21%20Do%20not%20do%20evil%20things%20with%20pimcore%20...%20%3B-%29
x-debug-exception-file: %2Fvar%2Fwww%2Fpimcore%2Fvendor%2Fpimcore%2Fpimcore%2Fbundles%2FAdminBundle%2FSecurity%2FCsrfProtectionHandler.php:68
cache-control: max-age=0, must-revalidate, private
x-debug-token: 244a31
x-debug-token-link: https://localhost/_profiler/244a31
x-robots-tag: noindex
x-powered-by: pimcore
content-language: en
pragma: no-cache
expires: Wed, 25 Aug 2021 06:28:08 GMT
x-previous-debug-token: 961c5c

**Comment by brusch** [comment]
@NiklasBr got the reason ... `curl -I https://localhost/admin` is doing a `HEAD` request, but the CSRF listener checks only for `GET`, see: 
https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/bundles/AdminBundle/EventListener/CsrfProtectionListener.php#L74

**Comment by NiklasBr** [comment]
The HEAD vs GET does not explain the 502 issue for admin pages unfortunately

```
$ curl https://localhost/admin --insecure
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <meta http-equiv="refresh" content="0;url='/admin/login?perspective='" />

        <title>Redirecting to /admin/login?perspective=</title>
    </head>
    <body>
        Redirecting to <a href="/admin/login?perspective=">/admin/login?perspective=</a>.
    </body>
</html>
$ curl https://localhost/admin/login?perspective= --insecure
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```
![image](https://user-images.githubusercontent.com/279826/130775708-30a3b185-64d5-4cb8-8008-c57563207448.png)

But the front end at `https://localhost/` is working fine. I can't find any differences in our configuration compared to [the docs](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Nginx_Configuration.html).

**Reply by brusch** [reply]
Hmm, for also nothing in the Nginx/FPM logs?

**Reply by NiklasBr** [reply]
FPM logs contain nothing (with the exception of e.g. deprecation notices, which also appear on the front end logs)
Nginx logs:
```
[error] 11#11: *41 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 10.199.100.29, server: localhost
```
So that we are sure about the configuration, in the docs there is a reference to 
```nginx
        # include fastcgi.conf if needed
        include fastcgi.conf;
```
Is it expected to be this file: https://github.com/nginx/nginx/blob/master/conf/fastcgi.conf ?

**Reply by NiklasBr** [reply]
Found the issue.

Changes to AbstractInstaller made a bundle do an infinite loop, crashing the entire process while not leaving any good traces in the log.
